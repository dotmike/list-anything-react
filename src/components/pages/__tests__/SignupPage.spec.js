import React from "react";
import configureStore from "redux-mock-store";

import ConnectedSignupPage, { UnconnectedSignupPage } from "../SignupPage";

const defaultProps = {
  history: {
    push: () => {}
  }
};

describe("ConnectedSignupPage", () => {
  const mockStore = configureStore();
  const initialState = {};
  const props = { ...defaultProps, store: mockStore(initialState) };

  it("renders correctly", () => {
    const wrapper = shallowWithContext(<ConnectedSignupPage {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("UnconnectedSignupPage", () => {
  const props = {
    ...defaultProps,
    signup: () => Promise.resolve()
  };

  it("renders correctly", () => {
    const wrapper = shallowWithContext(<UnconnectedSignupPage {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
