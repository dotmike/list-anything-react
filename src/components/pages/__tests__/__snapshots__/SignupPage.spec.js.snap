// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedSignupPage renders correctly 1`] = `
<SignupPage
  history={
    Object {
      "push": [Function],
    }
  }
  signup={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;

exports[`UnconnectedSignupPage renders correctly 1`] = `
<CenterElemsContainer>
  <Header
    as="h2"
    color="teal"
    textAlign="center"
  >
    Sign up now!
  </Header>
  <SignupForm
    submit={[Function]}
  />
  <Message>
    Do you already have an account? 
    <a
      href="/login"
    >
      Log in
    </a>
     now!
  </Message>
</CenterElemsContainer>
`;
